interfaces/builtin/unity8.go

```diff
@@ -40,6 +128,26 @@ func (iface *Unity8Interface) PermanentPlugSnippet(plug *interfaces.Plug, securi
 }
 
 func (iface *Unity8Interface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		oldTags := []byte("###SLOT_SECURITY_TAGS###")
+		newTags := slotAppLabelExpr(slot)
+		snippet := bytes.Replace([]byte(unity8ConnectedPlugAppArmor), oldTags, newTags, -1)
+
+		// FIXME: Until we decide whether unity8 is going to use Snappy-style
+		//        appIDs (snap.NAME.COMMAND) or Touch-style ones
+		//        (NAME_COMMAND_REVISION), we'll use a simpler *NAME* glob
+		//        here. This glob does not provide isolation because 'foo' can
+		//        access 'barfoobaz'. The base declaration restriction cannot
+		//        be lifted until this is properly mediated.
+		appidsOld := []byte("###PLUG_DBUS_APPIDS###")
+		appidsNew := fmt.Sprintf("*%s*", dbus.SafePath(plug.Snap.Name()))
```

Comment:
One option is to add an extra field in the interface declaration with the actual old-school application ID. We'll still need to keep reviews tight because it means snaps could impersonate different applications by tweaking this field alone, but that means we can properly tell what specific application ID is being shipped in the snap and being connected to unity8, and also gradually phase out that possibility if/when we decide to in the future.

@jdstrand Does that make sense, or am I missing something?